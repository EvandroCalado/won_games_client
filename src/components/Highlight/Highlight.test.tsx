import { render, screen } from '@testing-library/react';
import { Highlight } from '.';

const props = {
  title: 'Heading 1',
  subtitle: 'Heading 2',
  buttonLabel: 'Buy now',
  buttonLink: '/games/defy-death',
  backgroundImage: '/img/red-dead-img.jpg',
};

describe('Highlight', () => {
  it('should render headings and button', () => {
    render(<Highlight {...props} />);

    expect(
      screen.getByRole('heading', { name: /heading 1/i }),
    ).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /buy now/i })).toBeInTheDocument();
    expect(screen.getByRole('link', { name: /buy now/i })).toHaveAttribute(
      'href',
      '/games/defy-death',
    );
  });

  it('should render background image', () => {
    const { container } = render(<Highlight {...props} />);

    expect(container.firstChild).toHaveStyle({
      backgroundImage: `url(${props.backgroundImage})`,
    });
  });

  it('should render float image', () => {
    render(<Highlight {...props} floatImage="/float-image.png" />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      '/_next/image?url=%2Ffloat-image.png&w=3840&q=75',
    );
  });

  it('should render align right by default', () => {
    render(<Highlight {...props} floatImage="/float-image.png" />);

    expect(
      screen.getByRole('img', { name: props.title }).parentElement,
    ).not.toHaveClass('order-last');
  });

  it('should render align left', () => {
    render(
      <Highlight {...props} floatImage="/float-image.png" alignment="left" />,
    );

    expect(
      screen.getByRole('img', { name: props.title }).parentElement,
    ).toHaveClass('order-last');
  });

  it('should render a snapshot', () => {
    const { container } = render(
      <Highlight {...props} floatImage="/float-image.png" />,
    );

    expect(container).toMatchSnapshot();
  });
});
