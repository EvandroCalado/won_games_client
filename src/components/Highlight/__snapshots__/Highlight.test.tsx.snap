// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should render a snapshot 1`] = `
<div>
  <section
    class="relative grid h-56 grid-cols-6 bg-cover bg-center bg-no-repeat after:absolute after:h-full after:w-full after:bg-black/60 after:content-[''] max-md:-mx-4 md:h-80"
    style="background-image: url(/img/red-dead-img.jpg);"
  >
    <div
      class="false relative col-span-2 h-40 w-full self-end pl-4 sm:h-56 md:h-80 md:pl-11"
    >
      <img
        alt="Heading 1"
        class="z-20 object-contain"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="100vw"
        src="/_next/image?url=%2Ffloat-image.png&w=3840&q=75"
        srcset="/_next/image?url=%2Ffloat-image.png&w=640&q=75 640w, /_next/image?url=%2Ffloat-image.png&w=750&q=75 750w, /_next/image?url=%2Ffloat-image.png&w=828&q=75 828w, /_next/image?url=%2Ffloat-image.png&w=1080&q=75 1080w, /_next/image?url=%2Ffloat-image.png&w=1200&q=75 1200w, /_next/image?url=%2Ffloat-image.png&w=1920&q=75 1920w, /_next/image?url=%2Ffloat-image.png&w=2048&q=75 2048w, /_next/image?url=%2Ffloat-image.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
    </div>
    <div
      class="col-span-4 items-end z-10 flex flex-col gap-2 p-4 text-right md:justify-end md:p-11"
    >
      <h4
        class="text-2xl text-white font-bold md:text-3xl"
      >
        Heading 1
      </h4>
      <h6
        class="text-2xl text-white font-light md:text-xl"
      >
        Heading 2
      </h6>
      <a
        class="flex items-center justify-center gap-2 rounded w-max bg-gradient-to-b from-red to-primary p-2 text-white duration-150 hover:bg-gradient-to-b hover:from-red/95 hover:to-primary/95 active:bg-gradient-to-b active:from-red/80 active:to-primary/80 text-sm h-10 px-8 [&>svg]:w-5 [&>svg]:h-5 mt-6"
        href="/games/defy-death"
      >
        Buy now
      </a>
    </div>
  </section>
</div>
`;
